tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -258,12 +258,18 @@ func {{ template "copyFunctionName" $schema }}(ctx context.Context, s pgSearch.D
     if len(objs) == 0 {
         return nil
     }
-    batchSize := min(len(objs), pgSearch.MaxBatchSize)
-    inputRows := make([][]interface{}, 0, batchSize)
     {{if not $schema.Parent }}
-    // This is a copy so first we must delete the rows and re-add them
-    // Which is essentially the desired behaviour of an upsert.
-    deletes := make([]string, 0, batchSize)
+    {
+        // CopyFrom does not upsert, so delete existing rows first to achieve upsert behavior.
+        // Parent deletion cascades to children, so only the top-level parent needs deletion.
+        deletes := make([]string, 0, len(objs))
+        for _, obj := range objs {
+            deletes = append(deletes, {{ range $field := $schema.PrimaryKeys }}{{$field.Getter "obj"}}{{end}})
```

Comment:
Do we have an accessor to the object primary key ? If so, I would find its use less confusing than this `{{range ...}} ... {{ end}}` pattern.

Reply:
Fixed